### PR TITLE
Fix vs logs

### PR DESCRIPTION
**Pull request by thallesfreitaszup**
## Issue Description

Not saving logs when fails to create virtual service.

### CODE REVIEW

**Review by barbararochazup — CHANGES_REQUESTED** [review]
Update headers with 2021 year

File: butler/src/app/v2/operator/logs-aggregator/kubernetes-events-aggregator.ts [hunk 1]
```diff
@@ -69,17 +73,15 @@ export class EventsLogsAggregator {
 
     const deploymentId = this.getDeploymentIdLabel(resource)
 
+
     if (!deploymentId) {
       this.consoleLoggerService.log(`Resource ${involvedObject.kind}/${involvedObject.name} in namespace ${involvedObject.namespace} does not has label ${AppConstants.DEPLOYMENT_ID_LABEL}. Discarding event...`)
+      await this.checkResourceAnnotation(resource, event)
```
Comment (tiagoangelozup) [inline butler/src/app/v2/operator/logs-aggregator/kubernetes-events-aggregator.ts:79]: Maybe a wrapper of KubernetesObject could improve the code readability. 
e.g.
```ts
class ResourceWrapper {
  private readonly _resource: KubernetesObject
  private readonly _spec?: Record<string, unknown>

  constructor(resource: KubernetesObject) {
    this._resource = resource
    if (resource.kind === AppConstants.CHARLES_CUSTOM_RESOURCE_ROUTES_KIND) {
      this._spec = plainToClass(K8sManifestWithSpec, this).spec
    }
  }

  public get circles(): CharlesCircle[] {
    return this._spec ? this._spec.circles as CharlesCircle[] : []
  }

  public get deploymentId(): string | undefined {
    const deploymentId = this._resource.metadata?.labels?.[AppConstants.DEPLOYMENT_ID_LABEL]
    if (deploymentId) return deploymentId
    return this._spec ? this._spec.deploymentId as string : undefined
  }
}
```

File: butler/src/app/v2/operator/logs-aggregator/kubernetes-events-aggregator.ts [hunk 2]
```diff
@@ -88,8 +90,39 @@ export class EventsLogsAggregator {
...
+      const circles = spec ? spec.circles as CharlesCircle[] : null
+      const circlesIds = circles?.map(it => it.id)
+      if(circlesIds?.length){
+        const currentDeployments = await this.deploymentsRepository
+          .findCurrentsByCirclesIds(circlesIds)
+        const currentDeploymentsIds = currentDeployments
+          .map(it => it.id)
+        const log = this.createLogFromEvent(event)
+        await Promise.all(currentDeploymentsIds.map(
+          deploymentId => this.saveLogs(deploymentId, log)
```
Comment (tiagoangelozup) [inline butler/src/app/v2/operator/logs-aggregator/kubernetes-events-aggregator.ts:109]: I think the `LogRepository` is able to persist multiple entities at once with a single transaction.

**Review by tiagoangelozup — COMMENTED** [review]
Just a few suggestions

**Review by barbararochazup — APPROVED** [review]

**Review by tiagoangelozup — APPROVED** [review]

**Pull request merged**